### PR TITLE
accesslog: Enable L7 monitor logging by default

### PR DESCRIPTION
**Pull request by tgraf**
Fixes: #2850

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Review by rlenglet — APPROVED** [review]

**Review by ashwinp — APPROVED** [review]
lgtm

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: pkg/monitor/types.go [hunk 1]
```diff
@@ -31,3 +40,73 @@ const (
...
+
+var _ pflag.Value = &MessageTypeFilter{}
+
+func (m *MessageTypeFilter) String() string {
+	pieces := make([]string, 0, len(*m))
+	for _, typ := range *m {
+		pieces = append(pieces, type2name(typ))
+	}
+
+	return strings.Join(pieces, ", ")
```
Comment (rlenglet) [inline pkg/monitor/types.go]: I would remove the space, to make it easier to use. Like K8s label selector strings.

File: Documentation/cmdref/cilium_monitor.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ cilium monitor
       --hex                   Do not dissect, print payload in HEX
       --related-to []uint16   Filter by either source or destination endpoint id
       --to []uint16           Filter by destination endpoint id
-  -t, --type string           Filter by event types [capture debug drop trace]
+  -t, --type []int            Filter by event types [agent capture debug drop l7 trace]
```
Comment (rlenglet) [inline Documentation/cmdref/cilium_monitor.md]: `[]int` -> `[]string`

**Review by rlenglet — APPROVED** [review]
Just 2 nitpicks.

File: pkg/monitor/types.go (hunk 1 above)
Comment (tgraf) [reply pkg/monitor/types.go]: fixed

File: Documentation/cmdref/cilium_monitor.md (hunk 2 above)
Comment (tgraf) [reply Documentation/cmdref/cilium_monitor.md]: fixed

**Comment by tgraf** [comment]
test-me-please

File: cilium/cmd/monitor.go [hunk 3]
```diff
@@ -161,7 +142,7 @@ func lostEvent(lost uint64, cpu int) {
 // related to, which can match on both.  If either one of them is less than or
 // equal to zero, then it is assumed user did not use them.
 func match(messageType int, src uint16, dst uint16) bool {
-	if eventTypeIdx != monitor.MessageTypeUnspec && messageType != eventTypeIdx {
+	if len(eventTypes) > 0 && !eventTypes.Contains(messageType) {
```
Comment (raybejjani) [inline cilium/cmd/monitor.go:145]: .contains handles the len==0 (or nil) case, if you don't want to have 2 things.

File: pkg/proxy/accesslog/log.go [hunk 4]
```diff
@@ -78,15 +78,20 @@ type LogRecordNotifier interface {
 }
 
 // OpenLogfile opens a file for logging
-func OpenLogfile(lf string, n LogRecordNotifier) error {
+func OpenLogfile(lf string) error {
 	logMutex.Lock()
 	defer logMutex.Unlock()
 
-	notifier = n
```
Comment (raybejjani) [inline pkg/proxy/accesslog/log.go:85]: What was the motivation to make this a package global? It makes the relationship between OpenLogFile and the notifier more hidden and indirect. Perhaps there is another PR where we'll want to change the global notifier?
If you really want to keep it this way could you also make the comment on OpenLogfile explain what it does?

**Review by raybejjani — APPROVED** [review]

File: pkg/proxy/accesslog/log.go (hunk 4 above)
Comment (tgraf) [reply pkg/proxy/accesslog/log.go:85]: Happy about such a PR, this PR just splits the init code.

**Pull request merged**